diff --git a/src/osc/oscpack/osc/OscOutboundPacketStream.h b/src/osc/oscpack/osc/OscOutboundPacketStream.h
index 3cc2835..1a90a01 100644
--- a/src/osc/oscpack/osc/OscOutboundPacketStream.h
+++ b/src/osc/oscpack/osc/OscOutboundPacketStream.h
@@ -105,7 +105,7 @@ public:
     OutboundPacketStream& operator<<( const InfinitumType& rhs );
     OutboundPacketStream& operator<<( int32 rhs );
 
-#if !(defined(__x86_64__) || defined(_M_X64))
+#if !(defined(__x86_64__) || defined(_M_X64) || defined(__aarch64__))
     OutboundPacketStream& operator<<( int rhs )
             { *this << (int32)rhs; return *this; }
 #endif
diff --git a/src/osc/oscpack/osc/OscReceivedElements.h b/src/osc/oscpack/osc/OscReceivedElements.h
index b6205f3..69d0e6c 100644
--- a/src/osc/oscpack/osc/OscReceivedElements.h
+++ b/src/osc/oscpack/osc/OscReceivedElements.h
@@ -100,7 +100,7 @@ public:
         : contents_( contents )
         , size_( ValidateSize( (osc_bundle_element_size_t)size ) ) {}
 
-#if !(defined(__x86_64__) || defined(_M_X64))
+#if !(defined(__x86_64__) || defined(_M_X64) || defined(__aarch64__))
     ReceivedPacket( const char *contents, int size )
         : contents_( contents )
         , size_( ValidateSize( (osc_bundle_element_size_t)size ) ) {}
diff --git a/src/osc/oscpack/osc/OscTypes.h b/src/osc/oscpack/osc/OscTypes.h
index 61499ba..8077c30 100644
--- a/src/osc/oscpack/osc/OscTypes.h
+++ b/src/osc/oscpack/osc/OscTypes.h
@@ -49,7 +49,7 @@ namespace osc
 typedef __int64 int64;
 typedef unsigned __int64 uint64;
 
-#elif defined(__x86_64__) || defined(_M_X64)
+#elif defined(__x86_64__) || defined(_M_X64) || defined(__aarch64__)
 
 typedef int64_t int64;
 typedef uint64_t uint64;
@@ -61,7 +61,7 @@ typedef unsigned long long uint64;
 
 #endif
 
-#if defined(__x86_64__) || defined(_M_X64)
+#if defined(__x86_64__) || defined(_M_X64) || defined(__aarch64__)
 
 typedef signed int int32;
 typedef unsigned int uint32;
