diff --git a/src/engine/Engine.cpp b/src/engine/Engine.cpp
index 9e4b452f..78259583 100644
--- a/src/engine/Engine.cpp
+++ b/src/engine/Engine.cpp
@@ -25,7 +25,7 @@ inline void cpuPause() {
 #if defined ARCH_X64
 	_mm_pause();
 #elif defined ARCH_ARM64
-	__yield();
+	asm volatile("yield" ::: "memory");
 #endif
 }
 
diff --git a/arch.mk b/arch.mk
index 88b50819..8b402e52 100644
--- a/arch.mk
+++ b/arch.mk
@@ -6,6 +6,9 @@ ifneq (,$(findstring x86_64-,$(MACHINE)))
 else ifneq (,$(findstring arm64-,$(MACHINE)))
 	ARCH_ARM64 := 1
 	ARCH_CPU := arm64
+else ifneq (,$(findstring arm-,$(MACHINE)))
+	ARCH_ARM32 := 1
+	ARCH_CPU := arm32
 else
 $(error Could not determine CPU architecture of $(MACHINE))
 endif
diff --git a/dep.mk b/dep.mk
index 7db35fb5..211f83d5 100644
--- a/dep.mk
+++ b/dep.mk
@@ -13,8 +13,8 @@ DEP_FLAGS += -fPIC
 ifdef ARCH_X64
 	DEP_FLAGS += -march=nehalem
 endif
-ifdef ARCH_ARM64
-	DEP_FLAGS += -march=armv8-a+fp+simd
+ifdef ARCH_ARM32
+	DEP_FLAGS += -march=armv7 -mtune=cortex-a9 -mfpu=neon -mno-unaligned-access -DSIMDE_ENABLE_NATIVE_ALIASES
 endif
 
 ifdef ARCH_MAC
diff --git a/Makefile b/Makefile
index e1be07d1..80b56aa4 100644
--- a/Makefile
+++ b/Makefile
@@ -73,7 +73,7 @@ STANDALONE_SOURCES += adapters/standalone.cpp
 ifdef ARCH_LIN
 	STANDALONE_TARGET := Rack
 	STANDALONE_LDFLAGS += -static-libstdc++ -static-libgcc
-	STANDALONE_LDFLAGS += -Wl,-rpath=.
+	STANDALONE_LDFLAGS += -Wl,-rpath=. -latomic
 endif
 ifdef ARCH_MAC
 	STANDALONE_TARGET := Rack
diff --git a/compile.mk b/compile.mk
index a076c20d..1fabbf99 100644
--- a/compile.mk
+++ b/compile.mk
@@ -29,8 +29,8 @@ endif
 ifdef ARCH_X64
 	FLAGS += -march=nehalem
 endif
-ifdef ARCH_ARM64
-	FLAGS += -march=armv8-a+fp+simd
+ifdef ARCH_ARM32
+	FLAGS += -march=armv7 -mtune=cortex-a9 -mfpu=neon -mno-unaligned-access -DSIMDE_ENABLE_NATIVE_ALIASES
 endif
 
 ifdef ARCH_LIN
diff --git a/plugin.mk b/plugin.mk
index e2630f67..073ca3e1 100644
--- a/plugin.mk
+++ b/plugin.mk
@@ -32,7 +32,7 @@ ifdef ARCH_LIN
 	# When Rack loads a plugin, it symlinks /tmp/Rack2 to its system dir, so the plugin can link to libRack.
 	LDFLAGS += -Wl,-rpath=/tmp/Rack2
 	# Since the plugin's compiler could be a different version than Rack's compiler, link libstdc++ and libgcc statically to avoid ABI issues.
-	LDFLAGS += -static-libstdc++ -static-libgcc
+	LDFLAGS += -static-libstdc++ -static-libgcc -latomic
 	XDG_DATA_HOME ?= $(HOME)/.local/share
 	RACK_USER_DIR ?= $(XDG_DATA_HOME)/Rack2
 endif
@@ -60,6 +60,8 @@ all: $(TARGET)
 
 include $(RACK_DIR)/compile.mk
 
+FLAGS += -mfpu=neon -mno-unaligned-access
+
 clean:
 	rm -rfv build $(TARGET) dist
 
diff --git a/src/common.cpp b/src/common.cpp
index aad8aeab..9741f29d 100644
--- a/src/common.cpp
+++ b/src/common.cpp
@@ -33,9 +33,9 @@ const std::string APP_VERSION = TOSTRING(_RACK_VERSION);
 #if defined ARCH_X64
 	const std::string APP_CPU = "x64";
 	const std::string APP_CPU_NAME = "x64";
-#elif defined ARCH_ARM64
-	const std::string APP_CPU = "arm64";
-	const std::string APP_CPU_NAME = "ARM64";
+#elif defined ARCH_ARM32
+	const std::string APP_CPU = "arm32";
+	const std::string APP_CPU_NAME = "ARM32";
 #endif
 const std::string API_URL = "https://api.vcvrack.com";
 
diff --git a/include/arch.hpp b/include/arch.hpp
index 99cd27b0..924e31d6 100644
--- a/include/arch.hpp
+++ b/include/arch.hpp
@@ -7,6 +7,9 @@
 #if defined __aarch64__
 	#define ARCH_ARM64 1
 #endif
+#if defined __arm__
+	#define ARCH_ARM32 1
+#endif
 
 // Check OS
 #if defined _WIN32
diff --git a/dep/Makefile b/dep/Makefile
index f016299e..f0cee5d4 100755
--- a/dep/Makefile
+++ b/dep/Makefile
@@ -124,7 +124,8 @@ openssl-3.3.2:
 $(openssl): | openssl-3.3.2
 	# Set no-pinshared so OpenSSL doesn't call GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_PIN, ...) on initialization on Windows. This prevents libRack from being unloaded on Windows with FreeLibrary().
 	# Don't use no-shared because libcurl's configure uses the shared library to find OpenSSL.
-	cd openssl-3.3.2 && ./Configure --prefix="$(DEP_PATH)" --libdir=lib no-zlib no-capieng no-pinshared no-apps no-tests no-docs no-ui-console
+	# disable asm as otherwise it will try to compile armv8 asm, which will of course not work on armv7
+	cd openssl-3.3.2 && ./Configure --prefix="$(DEP_PATH)" --libdir=lib no-zlib no-capieng no-pinshared no-apps no-tests no-docs no-ui-console no-asm
 	$(MAKE) -C openssl-3.3.2
 	$(MAKE) -C openssl-3.3.2 install_sw
 
diff --git a/src/system.cpp b/src/system.cpp
index 190b6da3..cf49d7c2 100644
--- a/src/system.cpp
+++ b/src/system.cpp
@@ -934,45 +934,56 @@ void runProcessDetached(const std::string& path) {
 
 
 uint32_t getFpuFlags() {
-#if defined ARCH_X64
-	return _mm_getcsr();
-#elif defined ARCH_ARM64
-	uint64_t fpcr;
-	__asm__ volatile("mrs %0, fpcr" : "=r" (fpcr));
-	return fpcr;
-#endif
+//#if defined ARCH_X64
+//	return _mm_getcsr();
+//#elif defined ARCH_ARM64
+//	uint64_t fpcr;
+//	__asm__ volatile("mrs %0, fpcr" : "=r" (fpcr));
+//	return fpcr;
+//#endif
+	return 0;
 }
 
 void setFpuFlags(uint32_t flags) {
-#if defined ARCH_X64
-	_mm_setcsr(flags);
-#elif defined ARCH_ARM64
-	uint64_t fpcr = flags;
-	__asm__ volatile("msr fpcr, %0" :: "r" (fpcr));
-#endif
+//#if defined ARCH_X64
+//	_mm_setcsr(flags);
+//#elif defined ARCH_ARM64
+//	uint64_t fpcr = flags;
+//	__asm__ volatile("msr fpcr, %0" :: "r" (fpcr));
+//#endif
 }
 
 void resetFpuFlags() {
-	uint32_t flags = getFpuFlags();
-
-#if defined ARCH_X64
+//	uint32_t flags = getFpuFlags();
+//
+//#if defined ARCH_X64
+//	// Set CPU to flush-to-zero (FTZ) and denormals-are-zero (DAZ) mode
+//	// https://software.intel.com/en-us/node/682949
+//	// _MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);
+//	flags |= 0x8000;
+//	// _MM_SET_DENORMALS_ZERO_MODE(_MM_DENORMALS_ZERO_ON);
+//	flags |= 0x0040;
+//	// Round-to-nearest is default
+//	flags &= ~0x6000;
+//#elif defined ARCH_ARM64
+//	// Set Flush-to-Zero
+//	flags |= 1 << 24;
+//	// ARM64 always uses DAZ
+//	// Round-to-nearest is default
+//	flags &= ~((1 << 22) | (1 << 23));
+//#endif
+//
+//	setFpuFlags(flags);
+//	old cpu flag initialization taken from:
+//	https://github.com/VCVRack/Rack/commit/e2a4a6209d4ab84ece3e685b129b3300b8e370bf
+//	until i read up some some proper *FpuFlags implementation for armv7l
 	// Set CPU to flush-to-zero (FTZ) and denormals-are-zero (DAZ) mode
 	// https://software.intel.com/en-us/node/682949
-	// _MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);
-	flags |= 0x8000;
-	// _MM_SET_DENORMALS_ZERO_MODE(_MM_DENORMALS_ZERO_ON);
-	flags |= 0x0040;
-	// Round-to-nearest is default
-	flags &= ~0x6000;
-#elif defined ARCH_ARM64
-	// Set Flush-to-Zero
-	flags |= 1 << 24;
+	// On ARM64, this is a SIMDe function.
+	_MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);
 	// ARM64 always uses DAZ
-	// Round-to-nearest is default
-	flags &= ~((1 << 22) | (1 << 23));
-#endif
-
-	setFpuFlags(flags);
+	// Reset rounding mode to default (nearest)
+	std::fesetround(FE_TONEAREST);
 }
 
 
