diff --git a/compile.mk b/compile.mk
index dcc6af0d..4e453d5c 100644
--- a/compile.mk
+++ b/compile.mk
@@ -30,7 +30,7 @@ ifdef ARCH_X64
 	FLAGS += -march=nehalem
 endif
 ifdef ARCH_ARM64
-	FLAGS += -march=armv8-a+fp+simd
+	FLAGS += -march=armv8-a -mtune=cortex-a53 -DSIMDE_ENABLE_NATIVE_ALIASES
 endif
 
 ifdef ARCH_LIN
diff --git a/dep.mk b/dep.mk
index 7db35fb5..cd618b1c 100644
--- a/dep.mk
+++ b/dep.mk
@@ -14,7 +14,7 @@ ifdef ARCH_X64
 	DEP_FLAGS += -march=nehalem
 endif
 ifdef ARCH_ARM64
-	DEP_FLAGS += -march=armv8-a+fp+simd
+	DEP_FLAGS += -march=armv8-a -mtune=cortex-a53 -DSIMDE_ENABLE_NATIVE_ALIASES
 endif
 
 ifdef ARCH_MAC
diff --git a/dep/Makefile b/dep/Makefile
index 87c1278b..c234428a 100755
--- a/dep/Makefile
+++ b/dep/Makefile
@@ -181,6 +181,8 @@ libsamplerate-0.1.9:
 	rm libsamplerate-0.1.9.tar.gz
 
 $(libsamplerate): | libsamplerate-0.1.9
+	cp /usr/share/automake-1.17/config.guess libsamplerate-0.1.9/Cfg
+	cp /usr/share/automake-1.17/config.sub libsamplerate-0.1.9/Cfg
 	cd libsamplerate-0.1.9 && $(CONFIGURE) --disable-fftw --disable-sndfile
 	$(MAKE) -C libsamplerate-0.1.9
 	$(MAKE) -C libsamplerate-0.1.9 install
diff --git a/src/engine/Engine.cpp b/src/engine/Engine.cpp
index 9e4b452f..78259583 100644
--- a/src/engine/Engine.cpp
+++ b/src/engine/Engine.cpp
@@ -25,7 +25,7 @@ inline void cpuPause() {
 #if defined ARCH_X64
 	_mm_pause();
 #elif defined ARCH_ARM64
-	__yield();
+	asm volatile("yield" ::: "memory");
 #endif
 }
 
