diff --git a/arch.mk b/arch.mk
index 88b50819..a93cb92e 100644
--- a/arch.mk
+++ b/arch.mk
@@ -6,6 +6,9 @@ ifneq (,$(findstring x86_64-,$(MACHINE)))
 else ifneq (,$(findstring arm64-,$(MACHINE)))
 	ARCH_ARM64 := 1
 	ARCH_CPU := arm64
+else ifneq (,$(findstring aarch64-,$(MACHINE)))
+	ARCH_ARM64 := 1
+	ARCH_CPU := arm64
 else
 $(error Could not determine CPU architecture of $(MACHINE))
 endif
diff --git a/compile.mk b/compile.mk
index b72fd683..e167bc39 100644
--- a/compile.mk
+++ b/compile.mk
@@ -27,7 +27,7 @@ ifdef ARCH_X64
 endif
 ifdef ARCH_ARM64
 	FLAGS += -DARCH_ARM64
-	FLAGS += -march=armv8-a+fp+simd
+	FLAGS += -march=armv8-a -mtune=cortex-a53 -DSIMDE_ENABLE_NATIVE_ALIASES
 endif
 
 ifdef ARCH_LIN
@@ -91,7 +91,7 @@ build/%.m.o: %.m
 build/%.bin.o: %
 	@mkdir -p $(@D)
 ifdef ARCH_LIN
-	$(OBJCOPY) -I binary -O elf64-x86-64 -B i386:x86-64 --rename-section .data=.rodata,alloc,load,readonly,data,contents $< $@
+	$(OBJCOPY) -I binary -O elf64-littleaarch64 -B aarch64 --rename-section .data=.rodata,alloc,load,readonly,data,contents $< $@
 endif
 ifdef ARCH_WIN
 	$(OBJCOPY) -I binary -O pe-x86-64 -B i386:x86-64 --rename-section .data=.rodata,alloc,load,readonly,data,contents $< $@
diff --git a/dep.mk b/dep.mk
index 7db35fb5..cd618b1c 100644
--- a/dep.mk
+++ b/dep.mk
@@ -14,7 +14,7 @@ ifdef ARCH_X64
 	DEP_FLAGS += -march=nehalem
 endif
 ifdef ARCH_ARM64
-	DEP_FLAGS += -march=armv8-a+fp+simd
+	DEP_FLAGS += -march=armv8-a -mtune=cortex-a53 -DSIMDE_ENABLE_NATIVE_ALIASES
 endif
 
 ifdef ARCH_MAC
diff --git a/dep/Makefile b/dep/Makefile
index 87c1278b..c234428a 100755
--- a/dep/Makefile
+++ b/dep/Makefile
@@ -181,6 +181,8 @@ libsamplerate-0.1.9:
 	rm libsamplerate-0.1.9.tar.gz
 
 $(libsamplerate): | libsamplerate-0.1.9
+	cp /usr/share/automake-1.16/config.guess libsamplerate-0.1.9/Cfg
+	cp /usr/share/automake-1.16/config.sub libsamplerate-0.1.9/Cfg
 	cd libsamplerate-0.1.9 && $(CONFIGURE) --disable-fftw --disable-sndfile
 	$(MAKE) -C libsamplerate-0.1.9
 	$(MAKE) -C libsamplerate-0.1.9 install
diff --git a/src/engine/Engine.cpp b/src/engine/Engine.cpp
index 453a4a09..78b0a957 100644
--- a/src/engine/Engine.cpp
+++ b/src/engine/Engine.cpp
@@ -98,6 +98,9 @@ struct SpinBarrier {
 				return;
 #if defined ARCH_X64
 			__builtin_ia32_pause();
+#else
+// TODO: test if this is really required
+			asm volatile("yield" ::: "memory");
 #endif
 		}
 	}
@@ -147,6 +150,9 @@ struct HybridBarrier {
 				return;
 #if defined ARCH_X64
 			__builtin_ia32_pause();
+#else
+// TODO: test if this is really required
+			asm volatile("yield" ::: "memory");
 #endif
 		}
 
