diff --git a/compile.mk b/compile.mk
index 9580533c..040b4a54 100644
--- a/compile.mk
+++ b/compile.mk
@@ -82,7 +82,7 @@ build/%.m.o: %.m
 build/%.bin.o: %
 	@mkdir -p $(@D)
 ifdef ARCH_LIN
-	$(OBJCOPY) -I binary -O elf64-x86-64 -B i386:x86-64 --rename-section .data=.rodata,alloc,load,readonly,data,contents $< $@
+	$(OBJCOPY) -I binary -O elf32-i386 -B i386:i386 --rename-section .data=.rodata,alloc,load,readonly,data,contents $< $@
 endif
 ifdef ARCH_WIN
 	$(OBJCOPY) -I binary -O pe-x86-64 -B i386:x86-64 --rename-section .data=.rodata,alloc,load,readonly,data,contents $< $@
diff --git a/arch.mk b/arch.mk
index 88b50819..cd204fe3 100644
--- a/arch.mk
+++ b/arch.mk
@@ -1,8 +1,8 @@
 MACHINE := $(shell $(CC) -dumpmachine)
 
-ifneq (,$(findstring x86_64-,$(MACHINE)))
+ifneq (,$(findstring i686-,$(MACHINE)))
 	ARCH_X64 := 1
-	ARCH_CPU := x64
+	ARCH_CPU := i686
 else ifneq (,$(findstring arm64-,$(MACHINE)))
 	ARCH_ARM64 := 1
 	ARCH_CPU := arm64
diff --git a/src/common.cpp b/src/common.cpp
index 4f2869c0..cca2aecc 100644
--- a/src/common.cpp
+++ b/src/common.cpp
@@ -31,8 +31,8 @@ const std::string APP_VERSION = TOSTRING(_APP_VERSION);
 	const std::string APP_OS_NAME = "Linux";
 #endif
 #if defined ARCH_X64
-	const std::string APP_CPU = "x64";
-	const std::string APP_CPU_NAME = "x64";
+	const std::string APP_CPU = "i686";
+	const std::string APP_CPU_NAME = "i686";
 #elif defined ARCH_ARM64
 	const std::string APP_CPU = "arm64";
 	const std::string APP_CPU_NAME = "ARM64";
