diff --git a/Makefile b/Makefile
index 252310f..51ca89a 100644
--- a/Makefile
+++ b/Makefile
@@ -104,14 +104,14 @@ LDFLAGS += -lwinmm
 endif
 
 ifdef ARCH_LIN
-SOURCES += surge/src/linux/ConfigurationXml.S
+SOURCES += surge/src/linux/ConfigurationXmlARM.S
 LDFLAGS += -lstdc++fs -pthread
 FLAGS += -DUSE_STD_EXPERIMENTAL_FILESYSTEM
 
 # This is really a hack but...
-build/surge/src/linux/ConfigurationXml.S.o: surge/src/linux/ConfigurationXml.S
+build/surge/src/linux/ConfigurationXmlARM.S.o: surge/src/linux/ConfigurationXmlARM.S
 	mkdir -p build/surge/src/linux
-	cd surge/src/linux && $(CC) -c ConfigurationXml.S -o ../../../$@
+	cd surge/src/linux && $(CC) -c ConfigurationXmlARM.S -o ../../../$@
 
 endif
 
diff --git a/Makefile b/Makefile
index cb59d18..90563a0 100644
--- a/Makefile
+++ b/Makefile
@@ -21,7 +21,7 @@ FLAGS += -Isurge/src/common -Isurge/src/common/dsp \
 	-DTARGET_HEADLESS \
 	-DTARGET_RACK 
 
-FLAGS += -include immintrin.h
+#FLAGS += -include immintrin.h
 
 
 
