diff --git a/Makefile b/Makefile
index 3c996f8..f2a29a0 100644
--- a/Makefile
+++ b/Makefile
@@ -54,7 +54,7 @@ endif
 $(ffmpeg): $(lame)
 	# Don't use $(CONFIGURE) because this is a handwritten configure script
 	# ffmpeg bug: The pkgconfig dir is not set from pkgconfigdir at all. Set it with PKG_CONFIG_PATH instead.
-	cd ffmpeg && PKG_CONFIG_PATH="$(DEP_PATH)/lib/pkgconfig" ./configure --prefix="$(DEP_PATH)" --enable-pic --enable-gpl \
+	cd ffmpeg && PKG_CONFIG_PATH="$(DEP_PATH)/lib/pkgconfig" CPPFLAGS="-march=armv7 -mtune=cortex-a9 -mfpu=neon -mno-unaligned-access -DSIMDE_ENABLE_NATIVE_ALIASES" ./configure --prefix="$(DEP_PATH)" --enable-pic --enable-gpl \
 		--disable-programs --disable-doc --disable-avdevice --disable-swresample --disable-postproc --disable-avfilter --disable-network --disable-iconv --disable-alsa --disable-autodetect --disable-everything \
 		--enable-protocol=file \
 		$(FFMPEG_FORMATS)
