diff --git a/Makefile b/Makefile
index 2d535f9..bccb988 100644
--- a/Makefile
+++ b/Makefile
@@ -250,7 +250,7 @@ ifdef ARCH_MAC
 	# Perhaps inline assembly is used in libpd? Who knows.
 	cd dep/libpd && $(MAKE) MULTI=true STATIC=true ADDITIONAL_CFLAGS='-DPD_LONGINTTYPE="long long" $(DEP_MAC_SDK_FLAGS) -stdlib=libc++' ADDITIONAL_LDFLAGS='$(DEP_MAC_SDK_FLAGS) -stdlib=libc++'
 else
-	cd dep/libpd && $(MAKE) MULTI=true STATIC=true ADDITIONAL_CFLAGS='-DPD_LONGINTTYPE="long long"'
+	cd dep/libpd && $(MAKE) MULTI=true STATIC=true ADDITIONAL_CFLAGS='-DPD_LONGINTTYPE="long long" -DPD_INTERNAL'
 endif
 	cd dep/libpd && $(MAKE) install prefix="$(DEP_PATH)"
 endif
@@ -272,8 +272,12 @@ DISTRIBUTABLES += faust_libraries
 #LDFLAGS += -L/usr/local/lib -lfaust dep/lib/mir-gen.o dep/lib/mir.o
 
 $(libfaust):
-	cd dep && git clone "https://github.com/grame-cncm/faust.git" --recursive
-	cd dep/faust && git checkout 1dfc452a8250f3123b5100edf8c882e1cea407a1
+	# to make it compile with faust and supercollider both enabled this is required
+	# see: https://community.vcvrack.com/t/vult-on-vcv-prototype-works-again/19420/26
+	# and: https://community.vcvrack.com/t/vult-on-vcv-prototype-works-again/19420/31
+	cd dep && git clone "https://github.com/grame-cncm/faust.git"
+	cd dep/faust && git checkout 2.59.5
+	cd dep/faust && git submodule update --init --recursive
 	cd dep/faust/build && make cmake BACKENDS=interp.cmake TARGETS=interp.cmake
 	cd dep/faust/build && make install PREFIX="$(DEP_PATH)"
 	cp -r dep/faust/libraries/* faust_libraries/
