diff --git a/Makefile b/Makefile
index 8869e95..c5fee47 100755
--- a/Makefile
+++ b/Makefile
@@ -1,10 +1,8 @@
-# If RACK_DIR is not defined when calling the Makefile, defaults to use the Rack-SDK submodule
-RACK_DIR ?= Rack-SDK/lin
-
 include $(RACK_DIR)/arch.mk
 
+CXXFLAGS += -DLINK_PLATFORM_LINUX=1
+
 ifeq ($(ARCH_OS), lin)
-    RACK_DIR = Rack-SDK/lin
     CXXFLAGS += -DLINK_PLATFORM_LINUX=1
 endif
 
